### PR TITLE
get all jobs 

### PR DESCRIPTION
**Pull request by msalarab**
get all jobs and handle jobs screen ui state

### CODE REVIEW

File: app/src/main/java/com/cupcake/jobsfinder/data/remote/BaseResponse.kt
Comment (AmeerAmjed) [inline app/src/main/java/com/cupcake/jobsfinder/data/remote/BaseResponse.kt:1]: can use @Serializable with data class no need use **SerializedName** annotation to all fields

File: app/src/main/java/com/cupcake/jobsfinder/data/remote/response/job/JobDto.kt
Comment (AmeerAmjed) [inline app/src/main/java/com/cupcake/jobsfinder/data/remote/response/job/JobDto.kt:1]: can use @serializable with data class no need use SerializedName annotation to all fields

File: app/src/main/java/com/cupcake/jobsfinder/data/repository/Repository.kt
Comment (AmeerAmjed) [inline app/src/main/java/com/cupcake/jobsfinder/data/repository/Repository.kt:1]: in Repository return data fom dto form response in data layer

File: app/src/main/java/com/cupcake/jobsfinder/domain/usecase/GetAllJobUseCase.kt
Comment (AmeerAmjed) [inline app/src/main/java/com/cupcake/jobsfinder/domain/usecase/GetAllJobUseCase.kt:1]: where is mapper to **dto** to domain model

**Review by AmeerAmjed — CHANGES_REQUESTED** [review]

**Review by AmeerAmjed — APPROVED** [review]

**Review by TarekIdrees — APPROVED** [review]

**Pull request merged**